### PR TITLE
Port the T-ray scanner from Bay

### PR DESCRIPTION
**Pull request by TheGoret**
Basically took the t-ray scanner from Bay and adapted it to our codebase. Works fine and is available on 7x7 tiles. Flickering is gone too and people should use it more now. 

Also #7860

### CODE REVIEW

**Comment by TheGoret** [comment]
!review

File: code/game/objects/items/devices/t_scanner.dm [hunk 1]
```diff
@@ -9,27 +9,41 @@
...
+/obj/item/device/t_scanner/Destroy()
+	. = ..()
+	if(on)
+		set_active(FALSE)
+
 /obj/item/device/t_scanner/update_icon()
 	icon_state = "t-ray[on]"
 
+/obj/item/device/t_scanner/emp_act()
+	audible_message(src, "<span class = 'notice'> \The [src] buzzes oddly.</span>")
```
Comment (alsoandanswer) [inline code/game/objects/items/devices/t_scanner.dm]:
```suggestion
	audible_message(src, span("notice", "\The [src] buzzes oddly."))
```

File: code/game/objects/items/devices/t_scanner.dm [hunk 2]
```diff
@@ -9,27 +9,41 @@
...
+	set_active(FALSE)
+
 /obj/item/device/t_scanner/attack_self(mob/user)
 	set_active(!on)
+	user.update_action_buttons()
+
+/obj/item/device/t_scanner/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	var/obj/structure/disposalpipe/D = target
+	if(D && istype(D))
+		to_chat(user, "<span class='info'>Pipe segment integrity: [(D.health / 10) * 100]%</span>")
```
Comment (alsoandanswer) [inline code/game/objects/items/devices/t_scanner.dm]:
```suggestion
		to_chat(user, span("info", "Pipe segment integrity: [(D.health / 10) * 100]%"))
```

Comment (alsoandanswer) [reply code/game/objects/items/devices/t_scanner.dm]: in future change it to span proc

small change, but goes a long way in code cleanliness

**Pull request merged**